frontend/src/components/organisms/ViewEvents.tsx

```diff
@@ -83,55 +86,42 @@ const UpcomingEvents = () => {
   if (isLoading) return <Loading />;
 
   /** Error screen */
-  if (isError) return <Error />;
+  if (isError) return <FetchDataError />;
+
+  console.log(upcomingEventsSupervisor);
 
   return (
     <div>
-      {role === "Supervisor" ? (
-        <div>
-          <h2>
-            Yay!! you're a supervisor. You get special privileges on this page.
-          </h2>
-        </div>
-      ) : role === "Admin" ? (
-        <div>
-          <h2>
-            Yay!! you're an admin. You get special privileges on this page.
-          </h2>
-        </div>
-      ) : role === "Volunteer" ? (
-        <div>
-          <h2>You're a volunteer</h2>
-        </div>
-      ) : (
-        <></>
+      {(role === "Supervisor" || role === "Admin") && (
+        <Link href="/events/create">
+          <Button className="mb-2 w-full sm:w-max">Create New Event</Button>
+        </Link>
       )}
-      <Link href="/events/create">
-        <Button className="mb-2 w-full sm:w-max">Create New Event</Button>
-      </Link>
       {/* Display when no events are found */}
       {/* TODO: make this look better */}
-      {upcomingEventsSupervisor.length == 0 && upcomingEventsVolunteer == 0 && (
+      {((role === "Admin" && upcomingEventsSupervisor.length == 0) ||
+        (role === "Supervisor" && upcomingEventsSupervisor.length == 0) ||
+        (role === "Volunteer" && upcomingEventsVolunteer == 0)) && (
         <div className="p-10">
           <div className="text-center">You have no upcoming events</div>
         </div>
       )}
       {/* List of Upcoming events user supervises */}
-      {upcomingEventsSupervisor.map((event: ViewEventsEvent) => (
-        <div>
-          <div className="mt-5" />
-          <EventCardNew key={event.id} event={event} />
-        </div>
-      ))}
-
+      {(role === "Supervisor" || role === "Admin") &&
+        upcomingEventsSupervisor.map((event: ViewEventsEvent) => (
+          <div>
+            <div className="mt-5" />
+            <EventCardNew key={event.id} event={event} />
+          </div>
+        ))}
```

Comment:
I don't think the check for the role will be necessary here because the query for the `upcomingEventsSupervisor` is something like this `/events?ownerid=${userid}&date=upcoming&sort=startDate:desc`, notice the ownerid so the array only holds events a user created, which means it will be empty for volunteers?🤔

Reply:
That makes sense but the reason I actually implemented this was because I was logged in as an admin and I saw two event cards for the same event (one for volunteer and one for the supervisor/admin). I think it's better to be explicit to make sure it doesn't show up.

Reply:
Wait, I'm curious about this. It could be because a supervisor created an event, and also registered for it? Which speaks to one of the BUGS mentioned in this PR - [here](https://github.com/cornellh4i/lagos-volunteers/issues/222)

Reply:
I'm not really sure, it might be a possibility. I still think it's better to add a safeguard since it doesn't otherwise impact the functionality in any way.